src/components/activePanel/activePanel.service.js

```diff
@@ -2,25 +2,110 @@
+                        // start timer
+                        // call btn in hangup mode
+                        currentStatus.calls = ActivePanelStatusEnum.ACTIVE.enum;
+                        showActivePanelDrv();
+                        startTimer();
+                        //callBtnMode('hangup');
+                        break;
+
+                    case component === ActivePanelComponentEnum.CALLS.enum && status === ActivePanelStatusEnum.INACTIVE.enum :
+                        // component = call, status = inactive (hangup, disc')
+                        // stopTimer
+                        // call btn is in call mode
+                        // if screenShare is inactive, hide drv
+                        currentStatus.calls = ActivePanelStatusEnum.INACTIVE.enum;
+                        stopTimer();
+                        //callBtnMode('call');
+                        if (!isScreenSharingActive()) {
+                            hideActivePanelDrv();
+                        }
+                        break;
+
+                    case component === ActivePanelComponentEnum.SCREEN_SHARE.enum && status === ActivePanelStatusEnum.ACTIVE.enum :
+                        // component = screenShare, status = active
+                        // show drv
+                        // screenShare buttons are disabled
+                        currentStatus.screenSharing = ActivePanelStatusEnum.ACTIVE.enum;
+                        showActivePanelDrv();
+                        //screenShareMode(true);
+                        //screenShareBtnsMode('disabled');
+                        break;
+
+                    case component === ActivePanelComponentEnum.SCREEN_SHARE.enum && status === ActivePanelStatusEnum.INACTIVE.enum :
+                        // component = screenShare, status = inactive
+                        // check if call is active, if not hide drv
+                        // return shareScreen btns to enabled state
+                        currentStatus.screenSharing = ActivePanelStatusEnum.INACTIVE.enum;
+                        if (!isCallActive()) {
+                            hideActivePanelDrv();
+                        }
+                        //screenShareMode(false);
+                        //screenShareBtnsMode('enabled');
+                        break;
+
+                    default:
+                        hideActivePanelDrv();
+                        break;
+                }
+            };
+
             this.getActions = function () {
                 return actions;
             };
 
-            function _base(name, arg1) {
+            function _base(name, param1) {
                 var fn = actions[name];
                 if (angular.isFunction(fn)) {
-                    fn(arg1);
+                    fn(param1);
                 }
             }
 
-            this.showActivePanelDrv = _base.bind(null, 'showUI');
+            var showActivePanelDrv = _base.bind(null, 'showUI');
```

Comment:
Good thinking, better implementation will be iterating over the desired actions.